wgpu-hal/src/dx12/pipeline_desc.rs

```diff
@@ -0,0 +1,183 @@
+//! We try to use pipeline stream descriptors where possible, but this isn't allowed
+//! on some older windows 10 versions. Therefore, we also must have some logic to
+//! convert such descriptors to the traditional equivalent.
+//!
+//! The code for pipeline stream descriptors is extremely gross and extremely unsafe.
+//! It has been isolated to this file for that reason.
+
+use core::mem::ManuallyDrop;
+
+use alloc::vec::Vec;
+use windows::Win32::Graphics::Direct3D12;
+use windows::Win32::Graphics::Dxgi;
+
+impl super::RenderPipelineStateStreamDesc {
+    /// # Safety
+    ///
+    /// Returned bytes contain pointers into this struct, for them to be valid,
+    /// this struct not move or be dropped. As if `as_bytes<'a>(&'a self) -> Vec<u8> + 'a`
+    pub unsafe fn to_bytes(&self) -> Vec<u8> {
+        // This allocation is unpleasant but in general the struct can get large enough that
+        // an allocation isn't the worst thing in the world.
```

Comment:
similar to above comments, I don't think these handwavy statements are helpful. _of course_ an allocation sucks but _likely_ won't break us. But if there's something to say about this (maybe there isn't) then it should be about the tradoffs and why it's hard to avoid (is it? or has it just not been considered to move faster)
I get the sentiment, but this kind of stuff tends to hinder maintainability since it means to future readers - in the best case it's noise, in the worst case it hints at some context the author had which is lost now.